server/src/main.py

```diff
@@ -179,3 +168,53 @@ async def delete_file(workspace_id: str, path: str):
     # -- for a later date --
     # TODO: Check user has write permission for workspace
     # see @stat()
+
+
+@app.post("/workspaces/{workspace_id}/directory/{directory_path}")
+async def create_directory(workspace_id: str, directory_path: str):
+    empty_data = bytes()
+    data_stream = io.BytesIO(empty_data)
+    client.put_object(
+        workspace_id,
+        directory_path + "/",
+        data=data_stream,
+        length=0,
+        content_type="application/octet-stream",
+    )
+    return {"message": f"CREATED {directory_path} in {workspace_id}"}
+
+
+@app.delete(
+    "/workspaces/{workspace_id}/directory/{directory_path}",
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def delete_directory(workspace_id: str, directory_path: str):
+    errors_count: int = 0
+    del_list: list[str] = []
+
+    objects = list((client.list_objects(workspace_id, prefix=directory_path + "/")))
+
+    for i in objects:
+        logger.debug(f"Adding {i.object_name} to delete list")
+        del_list.append(DeleteObject(i.object_name))
+    errors = client.remove_objects(
+        workspace_id,
+        del_list,
+    )
+
+    logger.info(f"Added {len(del_list)} objects to be removed")
+
+    for error in errors:
+        errors_count += 1
+        logger.warning(
+            f"Failed to delete object '{error.name}' with error code '{error.code}'."
+        )
+
+    if errors_count != 0:
+        logger.warning(
+            f"FAILED to delete {errors_count} object(s) from {directory_path} in {workspace_id}"
+        )
+
+    logger.info(
+        f"DELETED {len(del_list) - errors_count} object(s) from {directory_path} in {workspace_id}. Tried to delete {len(del_list)} object(s)."
```

Comment:
I wonder if this format is a little easier to read in the log.

`Deleted 15 of 32 objects from {path} in {workspace}.`